src/core/selinux-access.h

```diff
@@ -6,15 +6,37 @@
 #include "bus-util.h"
 #include "manager.h"
 
-int mac_selinux_generic_access_check(sd_bus_message *message, const char *path, const char *permission, sd_bus_error *error);
+#define MAC_SELINUX_CLASS_SYSTEM            "system"
+#define MAC_SELINUX_CLASS_UNIT              "service"
+
+#define MAC_SELINUX_UNIT_PERM_ADDDEPENDENCY "modify"
+#define MAC_SELINUX_UNIT_PERM_DISABLE       "disable"
+#define MAC_SELINUX_UNIT_PERM_ENABLE        "enable"
+#define MAC_SELINUX_UNIT_PERM_LINK          "enable"
+#define MAC_SELINUX_UNIT_PERM_MASK          "disable"
+#define MAC_SELINUX_UNIT_PERM_PRESET        "modify"
+#define MAC_SELINUX_UNIT_PERM_REENABLE      "enable"
+#define MAC_SELINUX_UNIT_PERM_REVERT        "modify"
+#define MAC_SELINUX_UNIT_PERM_STATUS        "status"
+#define MAC_SELINUX_UNIT_PERM_UNMASK        "enable"
+
+struct mac_selinux_callback_userdata {
+        Manager *m;
+        sd_bus_message *message;
+        sd_bus_error *error;
+        const char *permission;
```

Comment:
hmm, these fields aren#t really seliux specific, but the last one. Maybe just fold this structure into the main one, and just do
```c
#ifdef HAVE_SELINUX
const char *selinux_permission;
#endif 
```

at the end?

Reply:
(or maybe even without the `#ifdef`)

Reply:
done